irony-cdb-json.el

```diff
@@ -48,6 +48,53 @@ source directory.")
     (get-compile-options (irony-cdb-json--get-compile-options))))
 
 ;;;###autoload
+(defun irony-cdb-json--find-common-root-directory (compile-commands-path)
+  "Return the common root directory for all source files listed in a cdb.
+
+Reads compilation database at COMPILE-COMMANDS-PATH, extracts all source file paths
+and returns the deepest directory which contains them all.
+
+Example:
+If cdb contains entries for \"/foo/bar/baz\" \"/foo/bar/baaz\" and \"/foo/bozze\"
+will return \"/foo\""
+  (let* ((files (mapcar 'car (irony-cdb-json--load-db (expand-file-name  compile-commands-path))))
+         (dirs (delete-dups (mapcar (lambda (x) (file-name-directory (expand-file-name x))) files )))
+         (dir-components (mapcar (lambda (dir)
+                                   (cl-remove-if (lambda (v) (string= "" v)) (split-string dir "/")))
+                                    dirs))
+         (minlen (apply 'min (mapcar 'length dir-components)))
+         (common-count (cl-loop for i in (number-sequence 0 minlen)
+                                while (= 1 (length (delete-dups (mapcar (lambda (val) (nth i val)) dir-components))))
+                                finally return i
+                                )))
+    (mapconcat 'file-name-as-directory (cons "/" (-take common-count (car dir-components))) "")
+    ))
+
+;;;###autoload
+(defun irony-cdb-json-add-compile-commands-path-auto-root (compile-commands-path)
```

Comment:
I lied. I'm :-1::-1:  on breaking someone's working setup. But it's your show, so I split this off into a separate PR. It's an independent change.